src/main/java/com/synopsys/integration/detect/tool/detector/inspectors/nuget/DotNetRuntimeAvailabilityVerifier.java

```diff
@@ -0,0 +1,63 @@
+package com.synopsys.integration.detect.tool.detector.inspectors.nuget;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.synopsys.integration.detectable.detectable.exception.DetectableException;
+
+public class DotNetRuntimeAvailabilityVerifier {
+    private static final String DOTNET_RUNTIME_PATTERN_VERSION_PLACEHOLDER = "<DOTNET_VERSION>";
+    private static final String DOTNET_RUNTIME_PATTERN_WITHOUT_VERSION = "Microsoft\\.(AspNetCore|NETCore){1}\\.(All|App){1}(\\s){1}"
+                                                                             + DOTNET_RUNTIME_PATTERN_VERSION_PLACEHOLDER
+                                                                             + "(\\.[0-9]+){0,2}(\\s){1}\\[.+\\]";
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private DotNetRuntimeFinder dotNetRuntimeFinder;
+
+    public DotNetRuntimeAvailabilityVerifier(DotNetRuntimeFinder dotNetRuntimeFinder) {
+        this.dotNetRuntimeFinder = dotNetRuntimeFinder;
+    }
+
+    public boolean isRuntimeAvailable(String semanticVersion) throws DetectableException {
+        String[] versionTokenStrings = StringUtils.split(semanticVersion, '.');
+        List<Integer> numericVersionTokens = Arrays.stream(versionTokenStrings)
+                                                 .filter(NumberUtils::isDigits)
+                                                 .map(NumberUtils::toInt)
+                                                 .collect(Collectors.toList());
+
+        if (numericVersionTokens.isEmpty()) {
+            logger.warn("Invalid semantic version parameter for dotnet runtime query");
+            return false;
+        }
+
+        Integer[] intArray = new Integer[numericVersionTokens.size()];
+        return isRuntimeAvailable(numericVersionTokens.toArray(intArray));
+    }
+
+    public boolean isRuntimeAvailable(Integer... versionTokens) throws DetectableException {
+        List<String> runtimes = dotNetRuntimeFinder.listAvailableRuntimes();
```

Comment:
I think this is improved, but I think we could isolate more if you take in a List<string> runtimes instead of taking the DotNetRuntimeFinder.

Reply:
The reason I was hesitant about doing something like that is that it would allow you to pass in an arbitrary list of Strings. Even if we wrapped those Strings in a new type of object (e.g. `DotNetRuntimeModel` or something), I think these are concerns that are already correctly coupled. Decoupling further would essentially make this a semVer regex util that just so happens to have a constant pattern related to dotnet runtimes. If we want to go that route, that's fine (obviously we would move the runtime pattern constant somewhere else), but I think that is what we're really talking about if we decouple this further.

Reply:
Gotcha, I can say it is very typical for parsers to take in a list of strings, such as [NpmParser](https://github.com/blackducksoftware/synopsys-detect/blob/6d3a5e0707e5f625b542719760a4b4692c3f11c9/detectable/src/main/java/com/synopsys/integration/detectable/detectables/npm/lockfile/parse/NpmLockfileParser.java#L56). And it is convention we know you can't just pass an arbitrary list of strings. That said I would not be opposed to a small data class like DotNetRuntimeResult(List<string> runtimes).

Personally not having to create a mock runtime executable and eliminating the explicit dependency outweigh protecting someone from passing wrong strings. But I would also be fine with what you have. Depends on the type of change that comes in next.